packages/gestalt/src/SideNavigation.js

```diff
@@ -102,46 +107,29 @@ export default function SideNavigation({
             mobileTitle={mobileTitle}
             id={id}
           >
-            {navigationChildren}
+            {children}
           </SideNavigationMobile>
         </ScrollBoundaryContainer>
       </SideNavigationProvider>
     );
   }
 
   return (
-    <SideNavigationProvider>
-      <ScrollBoundaryContainer>
-        <Box minWidth={280} width={280} height="100%" as="nav" aria-label={accessibilityLabel}>
-          <div
-            className={
-              showBorder ? classnames(borderStyles.borderRight, styles.fullHeight) : undefined
-            }
-          >
-            <Box
-              padding={2}
-              color="default"
-              dangerouslySetInlineStyle={{ __style: { paddingBottom: 24 } }}
-              height="100%"
-            >
-              <Flex direction="column" gap={{ column: 4, row: 0 }}>
-                {header ? (
-                  <Flex direction="column" gap={{ column: 4, row: 0 }}>
-                    <Box paddingX={4}>{header}</Box>
-                    <Divider />
-                  </Flex>
-                ) : null}
-                <ul className={classnames(styles.ulItem)}>{navigationChildren}</ul>
-                {footer ? (
-                  <Flex direction="column" gap={{ column: 4, row: 0 }}>
-                    <Divider />
-                    <Box paddingX={4}>{footer}</Box>
-                  </Flex>
-                ) : null}
-              </Flex>
-            </Box>
-          </div>
-        </Box>
+    <SideNavigationProvider
+      collapsible={collapsible}
+      collapsed={collapsed}
+      onCollapse={onCollapse}
+      onPreview={onPreview}
+    >
+      <ScrollBoundaryContainer overflow={collapsible ? 'visible' : undefined}>
```

Comment:
We have to make overflow visible, because if not, box shadow in preview state will not be visible.

Also `ScrollBoundaryContainer` is soon to be deprecated and removed.

Reply:
I think it's ok. But we need to be aware we might run into issues:

![Brave Browser - SideNavigation - Gestalt 2024-03-13 at 11 42 14 AM](https://github.com/pinterest/gestalt/assets/10593890/33b42137-7a07-49d7-ae99-de83ca5b682f)